mock-rollout/src/store/HardwareContext.tsx

```diff
@@ -2,37 +2,41 @@ import React, { createContext, FC, useEffect, useState, useReducer } from 'react
 import { IHardware } from '../model/HardwareModel';
 import { HardwareService } from '../services/HardwareService';
 
-interface IState {
-    hardwares?: IHardware[] | null | undefined;
+interface IHardwareContextState {
+    hardwares?: IHardware[];
     error?: string;
+}
+
+interface IHardwareContext {
+    hardwareState: IHardwareContextState,
+    dispatchHardwareState?: (action: IAction) => void,
     updateHardware?: (id: number, hardware: IHardware) => void
 }
 
-interface IHardwareProps  {
+interface IHardwareProviderState {
     children: React.ReactNode;
 }
 
 type IAction = 
-    | { type: 'GET_SUCCESS', hardwares: IHardware[] | null, error: string }
-    | { type: 'GET_FAILED', hardwares: IHardware[] | null, error: string }
+    | { type: 'GET_SUCCESS', hardwares: IHardware[], error: string }
+    | { type: 'GET_FAILED', hardwares: IHardware[], error: string }
     | { type: 'EDIT', hardwares: IHardware[] }
 
 
-const initialHardwareState: IState = {
+const initialHardwareState: IHardwareContextState = {
     hardwares: [],
-    error: '',
-    updateHardware: (id: number, hardware: IHardware) => {}
+    error: ''
+}
+
+const initialContextState: IHardwareContext = {
+    hardwareState: initialHardwareState,
+    dispatchHardwareState: undefined,
+    updateHardware: undefined
 }
 
-export const HardwareContext = createContext<{
```

Comment:
you tried here with the Context state but it was too abstract to separate between those types of functions